### PR TITLE
Replace "student" with "learner".

### PR DESCRIPTION
**Pull request by dianakhuang**
## [TNL-3629](https://openedx.atlassian.net/browse/TNL-3629)

Switch "student" to "learner" in user-facing strings and documentation. The docstrings and class names that show up in the developer documentation have not been changed, but the general developer docs have been switched over to using "learner" instead of "student".

### Sandbox
- [x] http://dianakhuang.sandbox.edx.org/

### Testing
- [ ] ~~i18n~~ will be a separate translations-only PR
- [x] Unit, integration, acceptance tests as appropriate

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @andy-armstrong 
- [x] Code review: @efischer19 
- [x] UI strings/error msgs review: @catong 
- [x] Product review: @explorerleslie 

### Post-review
- [x] Squash commits

### CODE REVIEW

**Comment by catong** [comment]
@dianakhuang All the developer doc can be deleted also; thanks!

File: openassessment/templates/openassessmentblock/edit/oa_edit.html [hunk 1]
```diff
@@ -119,7 +119,7 @@ <h2 class="openassessment_alert_title">{% trans "Rubric Change Impacts Settings
                             <option value="1" {% if allow_latex %} selected="true" {% endif %}>{% trans "True"%}</option>
                         </select>
                     </div>
-                    <p class="setting-help">{% trans "Specify whether students can write Latex formatted strings"%}</p>
+                    <p class="setting-help">{% trans "Specify whether learners can write Latex formatted strings"%}</p>
```
Comment (catong) [inline openassessment/templates/openassessmentblock/edit/oa_edit.html]: "Latex" -> "LaTeX"

File: openassessment/templates/openassessmentblock/edit/oa_edit.html [hunk 2]
```diff
@@ -139,9 +139,9 @@ <h2 class="openassessment_alert_title">{% trans "Rubric Change Impacts Settings
 
             <p class="openassessment_description" id="openassessment_step_select_description">
                 {% if 'example_based_assessment' in editor_assessments_order %}
-                    {% trans "In this assignment, you can include steps for student training, peer assessment, self assessment, and example based assessment. Select the steps that you want below, and then drag them into the order that you want. If you include an example based assessment step, it must precede all other steps. If you include a student training training step, it must precede peer and self assessment steps." %}
+                    {% trans "In this assignment, you can include steps for learner training, peer assessment, self assessment, and example based assessment. Select the steps that you want below, and then drag them into the order that you want. If you include an example based assessment step, it must precede all other steps. If you include a learner training training step, it must precede peer and self assessment steps." %}
```
Comment (catong) [inline openassessment/templates/openassessmentblock/edit/oa_edit.html]: In "If you include a learner training training step", the word "training" is repeated; delete one of the occurrences.

File: openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html [hunk 3]
```diff
@@ -38,7 +38,7 @@
                     <textarea id="openassessment_rubric_feedback" class="input setting-input">{{ feedbackprompt }}</textarea>
                 </div>
                 <p class="setting-help">
-                    {% trans "Encourage your students to provide feedback on the response they've graded. You can replace the sample text with your own." %}
+                    {% trans "Encourage your learners to provide feedback on the response they've graded. You can replace the sample text with your own." %}
```
Comment (catong) [inline openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html]: "they've" -> "they have"

**Comment by dianakhuang** [comment]
@catong I've removed all the documentation and updated the problematic strings. Let me know if you think I need to do anything else.

**Comment by efischer19** [comment]
Seems straightforward enough, :+1: from me.

I think keeping all 3 commits separate is a good idea, since they each do different things the commit history will be useful down the road.

**Comment by catong** [comment]
Thanks @dianakhuang I am just looking at the sandbox to try to see some of the strings in context.

File: openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html [hunk 4]
```diff
@@ -38,7 +38,7 @@
                     <textarea id="openassessment_rubric_feedback" class="input setting-input">{{ feedbackprompt }}</textarea>
                 </div>
                 <p class="setting-help">
-                    {% trans "Encourage your students to provide feedback on the response they've graded. You can replace the sample text with your own." %}
+                    {% trans "Encourage your learners to provide feedback on the response they have graded. You can replace the sample text with your own." %}
```
Comment (andy-armstrong) [inline openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html]: Nit: I find the phrase "Encourage your learners..." to be a little strange. It works for me as "Encourage your students..." but I don't know that the authors feel that they own the learners in a course. :-) FYI @catong

File: openassessment/xblock/defaults.py [hunk 5]
```diff
@@ -75,7 +75,7 @@
 DEFAULT_EXAMPLE_ANSWER = (
     "Replace this text with your own sample response for this assignment. "
     "Then, under Response Score to the right, select an option for each criterion. "
-    "Students learn to assess responses by assessing this response and comparing "
+    "Learners learn to assess responses by assessing this response and comparing "
```
Comment (andy-armstrong) [inline openassessment/xblock/defaults.py]: Nit: "Learners learn..." is a little clumsy... @catong

File: openassessment/templates/openassessmentblock/staff_debug/student_info.html [hunk 6]
```diff
@@ -236,6 +236,6 @@ <h3 class="title">{% trans "Example-Based Assessment" %}</h3>
         {% endif %}
     </div>
     {% else %}
-        {% trans "Couldn't find a response for this student." %}
+        {% trans "Couldn't find a response for this learner." %}
```
Comment (andy-armstrong) [inline openassessment/templates/openassessmentblock/staff_debug/student_info.html]: Nit: I believe that our style guide discourages contractions. @catong, is that right?

File: openassessment/xblock/staff_info_mixin.py [hunk 7]
```diff
@@ -376,9 +376,9 @@ def cancel_submission(self, data, suffix=''):
                 cancelled_by_id=student_item_dict['student_id'],
                 assessment_requirements=assessment_requirements
             )
-            return {"success": True, 'msg': self._(u"The student submission has been removed from peer assessment. "
-                                                   u"The student receives a grade of zero unless you reset "
-                                                   u"the student's attempts for the problem to allow them to "
+            return {"success": True, 'msg': self._(u"The learner submission has been removed from peer assessment. "
```
Comment (andy-armstrong) [inline openassessment/xblock/staff_info_mixin.py]: Nit: fix this non-standard Python indentation.

File: openassessment/xblock/student_training_mixin.py [hunk 8]
```diff
@@ -50,7 +50,7 @@ def render_student_training(self, data, suffix=''):   # pylint:disable=W0613
         try:
             path, context = self.training_path_and_context()
         except: # pylint:disable=W0702
-            msg = u"Could not render student training step for submission {}".format(self.submission_uuid)
+            msg = u"Could not render learner training step for submission {}".format(self.submission_uuid)
```
Comment (andy-armstrong) [inline openassessment/xblock/student_training_mixin.py]: Nit: there is an inconsistency as to the capitalization of "Learner Training step". Most other usages in this PR go with the capitalization.

File: openassessment/xblock/student_training_mixin.py [hunk 9]
```diff
@@ -185,8 +185,8 @@ def training_assess(self, data, suffix=''): # pylint:disable=W0613
             )
         except student_training.StudentTrainingRequestError:
             msg = (
-                u"Could not check student training scores for "
-                u"the student with submission UUID {uuid}"
+                u"Could not check learner training scores for "
```
Comment (andy-armstrong) [inline openassessment/xblock/student_training_mixin.py]: Nit: I'd pull this back onto a single line.

**Comment by andy-armstrong** [comment]
:+1: Nice clean up, @dianakhuang. I have a few nits but nothing serious.

File: openassessment/templates/openassessmentblock/staff_debug/student_info.html (hunk 6 above)
Comment (catong) [inline openassessment/templates/openassessmentblock/staff_debug/student_info.html]: @andy-armstrong Yes, we discourage contractions. I was wrestling with the strange wording in this message. The least of many evils might be to go with passive voice and say:
"A response was not found for this learner."
@dianakhuang Sorry, can we make this change too?

File: openassessment/templates/openassessmentblock/student_training/student_training_error.html [hunk 10]
```diff
@@ -6,7 +6,7 @@
         <div class="wrapper--step__content">
 
             <div class="step__message message message--incomplete">
-                <h3 class="message__title">{% trans "Error Loading Student Training Examples" %}</h3>
+                <h3 class="message__title">{% trans "Error Loading Learner Training Examples" %}</h3>
 
                 <div class="message__content">
                     <p>{% trans "We couldn't load the student training step of this assignment." %}</p>
```
Comment (catong) [inline openassessment/templates/openassessmentblock/student_training/student_training_error.html]: @dianakhuang In line with @andy-armstrong 's suggestion to remove the contraction in messages elsewhere, can we change this string to: "The Learner Training Step for this assignment could not be loaded." Thanks!

File: openassessment/xblock/defaults.py [hunk 11]
```diff
@@ -75,7 +75,7 @@
...
     "the options that they select in the rubric with the options that you specified."
 )
```
Comment (catong) [inline openassessment/xblock/defaults.py]: @andy-armstrong @dianakhuang How about: "Learners practice performing peer assessments by assessing this response and comparing..."

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js [hunk 12]
```diff
@@ -95,7 +95,7 @@ OpenAssessment.StudentTrainingListener.prototype = {
         if (criterionAdded) {
             this.displayAlertMsg(
                 gettext("Criterion Added"),
-                gettext("You've added a criterion. You'll need to select an option for the criterion in the Student Training step. To do this, click the Settings tab.")
+                gettext("You've added a criterion. You'll need to select an option for the criterion in the Learner Training step. To do this, click the Settings tab.")
```
Comment (catong) [inline openassessment/xblock/static/js/src/studio/oa_edit_listeners.js]: "You've added..." -> "You have added..."
"You'll need to select..." -> "You will need to select..."

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js [hunk 13]
```diff
@@ -142,7 +142,7 @@ OpenAssessment.StudentTrainingListener.prototype = {
         if (invalidated) {
             this.displayAlertMsg(
                 gettext("Option Deleted"),
-                gettext("You've deleted an option. The system has removed that option from its criterion in the sample responses in the Student Training step. You may have to select a new option for the criterion.")
+                gettext("You've deleted an option. The system has removed that option from its criterion in the sample responses in the Learner Training step. You may have to select a new option for the criterion.")
```
Comment (catong) [inline openassessment/xblock/static/js/src/studio/oa_edit_listeners.js]: "You've deleted..." -> "You have deleted..."
"The system has removed that option..." -> "That option has been removed from its criterion..."
"You may have to select..." -> "You might have to select..."

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js [hunk 14]
```diff
@@ -174,7 +174,7 @@ OpenAssessment.StudentTrainingListener.prototype = {
         if (changed) {
             this.displayAlertMsg(
                 gettext("Option Deleted"),
-                gettext("You've deleted all the options for this criterion. The system has removed the criterion from the sample responses in the Student Training step.")
+                gettext("You've deleted all the options for this criterion. The system has removed the criterion from the sample responses in the Learner Training step.")
```
Comment (catong) [inline openassessment/xblock/static/js/src/studio/oa_edit_listeners.js]: "You've deleted..." -> "You have deleted..."
"The criterion has been removed from the example responses in the Learner Training step." 
(note change from "sample" to "example")

File: openassessment/xblock/static/js/src/studio/oa_edit_listeners.js [hunk 15]
```diff
@@ -201,7 +201,7 @@ OpenAssessment.StudentTrainingListener.prototype = {
         if (changed) {
             this.displayAlertMsg(
                 gettext("Criterion Deleted"),
-                gettext("You've deleted a criterion. The system has removed the criterion from the sample responses in the Student Training step.")
+                gettext("You've deleted a criterion. The system has removed the criterion from the sample responses in the Learner Training step.")
```
Comment (catong) [inline openassessment/xblock/static/js/src/studio/oa_edit_listeners.js]: "You've deleted" -> "You have deleted"
"The system has removed..." -> "The criterion has been removed from the example responses..."
(note change from "sample responses" to "example responses".)

File: openassessment/xblock/student_training_mixin.py [hunk 16]
```diff
@@ -49,8 +49,8 @@ def render_student_training(self, data, suffix=''):   # pylint:disable=W0613
 
         try:
             path, context = self.training_path_and_context()
-        except: # pylint:disable=W0702
-            msg = u"Could not render student training step for submission {}".format(self.submission_uuid)
+        except:  # pylint:disable=W0702
+            msg = u"Could not render Learner Training step for submission {}".format(self.submission_uuid)
```
Comment (catong) [inline openassessment/xblock/student_training_mixin.py]: @dianakhuang I am not able to get to a state where I see these messages in the sandbox. I'm not clear how they render in the UI. Is it odd or OK that these strings do not end with a period or other punctuation at the end of the sentence?

File: openassessment/xblock/test/test_staff_info.py [hunk 17]
```diff
@@ -102,7 +102,7 @@ def test_course_student_debug_info(self, xblock):
         # If we ARE course staff, then we should see the debug info
         xblock.xmodule_runtime.user_is_staff = True
         resp = self.request(xblock, 'render_student_info', json.dumps({}))
-        self.assertIn("couldn\'t find a response for this student.", resp.decode('utf-8').lower())
+        self.assertIn("couldn\'t find a response for this learner.", resp.decode('utf-8').lower())
```
Comment (catong) [inline openassessment/xblock/test/test_staff_info.py]: Are these user-visible strings? Not sure if we need to fix punctuation, capitalization.

File: openassessment/xblock/validation.py [hunk 18]
```diff
@@ -293,7 +293,7 @@ def validate_assessment_examples(rubric_dict, assessments, _):
 
             # Must have at least one training example
             if len(examples) == 0:
-                return False, _(u"Student training and example-based assessments must have at least one training example")
+                return False, _(u"Learner training and example-based assessments must have at least one training example")
```
Comment (catong) [inline openassessment/xblock/validation.py]: Missing period at end of sentence?

File: openassessment/xblock/student_training_mixin.py (hunk 16 above)
Comment (dianakhuang) [inline openassessment/xblock/student_training_mixin.py]: These are errors messages recorded in the system logs. I'll add a period to the end of the sentence.

File: openassessment/xblock/test/test_staff_info.py (hunk 17 above)
Comment (dianakhuang) [inline openassessment/xblock/test/test_staff_info.py]: This is a test, so it's okay if it's only checking for sentence fragments.

File: openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html (hunk 4 above)
Comment (dianakhuang) [inline openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html]: @catong is there anything we should do about this sentence?

**Comment by catong** [comment]
Thanks @dianakhuang! Looks great :+1:

File: openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html (hunk 4 above)
Comment (catong) [inline openassessment/templates/openassessmentblock/edit/oa_edit_rubric.html]: How about just removing the "your": "Encourage learners to provide feedback..."

**Pull request merged**